src/Raygun4php/RaygunClient.php

```diff
@@ -540,8 +537,8 @@ function getFilterParams() {
     /**
      * Use a proxy for sending HTTP requests to Raygun.
      *
-     * @param String $url URL including protocol and an optional port, e.g. http://myproxy:8080
-     * @return $this
+     * @param string $proxy URL including protocol and an optional port, e.g. http://myproxy:8080
+     * @return self
```

Comment:
I'm curious about this - does phpstan have a problem with `@return $this`?

Reply:
docs say

> Syntax
>     @return [Type] [<description>]

https://docs.phpdoc.org/references/phpdoc/tags/return.html

and `self` is a keyword

https://docs.phpdoc.org/references/phpdoc/types.html

So PHPDoc and phpstan agree.

Reply:
True, I'm not disputing the legitimacy of `@return self`, but questioning if `@return $this` is invalid. We use `@return $this` extensively in SilverStripe code, which is why I'm asking =)

Reply:
`$this` is an actual object, not a Type

Reply:
I see - PSR-5 (in draft) supports $this ([see here](https://github.com/php-fig/fig-standards/blob/master/proposed/phpdoc.md#abnf)) which is why I think it has been widely adopted. What you've changed it to is fine though.

Reply:
> I see - PSR-5 (in draft) supports $this as a Type

I see :) There are bad part of standards.

Reply:
It's a bit of a bikeshed colour, but "self" is less surprising than "$this" as a type to me. I note that szepeviktor has recommended a bulk change to framework as part of https://github.com/silverstripe/silverstripe-framework/pull/9103

Reply:
I don't personally mind, but I'd note that we've widely adopted `@return $this` and that it is part of PSR-5 (the new PHPDoc standard which is proposed but not accepted yet)

Reply:
I know there are people and standards and computers but `$this` does not resemble a type.

Reply:
Agreed, but does it need to? 😄Again, not opposed to changing SilverStripe's policy on how we document this, but the necessity for `@return` to be a Type is part of a specification (which PHPDoc created) and it is super old - PSR-5 is brand new (still not accepted by PHP-FIG) and says it can be a Type or `$this`